### DISCUSSION TITLE
Is it a supported feature that `on_translated` hook can modify the old/new list (of `_Action`) to change the output?

### DISCUSSION THREAD

**Discussion by user202729** [opensteno/plover#1297] [Plugin Support] [answered]
... as used by the `plover_fancytext` plugin.

I think this is a good feature but...

It would be better if the hook can receive the list of previous translations, so it doesn't have to keep track of them manually (if it wants to undo more than that in the `old` list).

(currently there's no hook documentation...)

**Comment by benoit-pierre** [comment, marked as answer]
No.

**Reply by user202729** [reply]
Thinking about it a little more, editing individual `_Action` objects is somewhat-okay (only in the `new` list), but modifying the list objects would be problematic, because they're not bound to the `Translation` objects.

In any case -- it's better to have plugins that works in a particular version, than to not have the plugin at all.
